identity/src/ecdsa.rs

```diff
@@ -285,4 +308,12 @@ mod tests {
         let invalid_msg = "h3ll0 w0rld".as_bytes();
         assert!(!pk.verify(invalid_msg, &sig));
     }
+
+    #[test]
+    fn secret_key_encode_decode_roundtrip() {
```

Comment:
Can you replace this test with a roundtrip test for an ECDSA key using the `Keypair::to_protobuf_encoding`? Plus one "test vector", i.e. parse hex-encoded protobuf bytes and assert that we can generic a particular `Peerid` from it.

The same way we did it in https://github.com/libp2p/rust-libp2p/pull/3681.

Reply:
I guess that's what you expect with `1c8d282`?

Reply:
Yep, thanks! I adapted the test vectors to actually be correct.